pkg/btfhelpers/btfhelpers.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package btfhelpers
+
+import (
+	"reflect"
+
+	"github.com/cilium/ebpf/btf"
+)
+
+func GetType(typ btf.Type) (reflect.Type, []string) {
+	switch typedMember := typ.(type) {
+	case *btf.Array:
+		arrType, typeNames := getSimpleType(typedMember.Type)
+		if arrType == nil {
+			return nil, nil
+		}
+		return reflect.ArrayOf(int(typedMember.Nelems), arrType), append(typeNames, typ.TypeName())
+	default:
+		arrType, typeNames := getSimpleType(typ)
+		return arrType, append(typeNames, typ.TypeName())
+	}
+}
+
+func GetUnderlyingType(tf *btf.Typedef) (btf.Type, []string, error) {
```

Comment:
Function `getUnderlyingType` already exists in two copies in the main branch:
```
pkg/gadgets/run/tracer/run.go:295:func getUnderlyingType(tf *btf.Typedef) (btf.Type, error) {
pkg/gadgets/run/types/metadata.go:359:func getUnderlyingType(tf *btf.Typedef) (btf.Type, error) {
```

So it can be good to move it in a new package and export it. But you can remove the old implementation so that we don't have 3 copies, and make a PR with just that change.

Reply:
The new version is slightly different (returns the tags) - but yeah, this can be extracted. 😄

Reply:
Moved to #2521